src/core/namespace.c

```diff
@@ -2486,7 +2486,7 @@ int setup_namespace(
                 goto finish;
 
         /* MS_MOVE does not work on MS_SHARED so the remount MS_SHARED will be done later */
-        r = mount_pivot_root(root);
+        r = mount_pivot_root(root, PROPAGATION_INHERIT);
```

Comment:
Then, you can simply pass `mount_flags`, and `mount()` below can be dropped.

Reply:
In that case I'd just open-code mount_move_root() completely and probably also rename mount_pivot_root()